### PR TITLE
chore: add ESM build

### PR DESCRIPTION
**Pull request by layershifter**
This PR:
- bumps Babel and Jest to latest versions
- adds a ESM build

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1 +1,14 @@
-{ "presets": ["env", "flow"] }
+{
+  "presets": [
+    ["@babel/preset-env", { "modules": false }],
+    "@babel/preset-flow"
+  ],
+  "env": {
+    "cjs": {
+      "presets": [["@babel/preset-env", { "targets": { "node": 6 } }]]
```
Comment (probablyup) [inline .babelrc:8]: this can be bumped up to 10, the current LTS version

File: package.json [hunk 2]
```diff
@@ -2,9 +2,12 @@
   "name": "stylis-plugin-rtl",
   "version": "1.0.0",
   "description": "Fork of stylis-rtl, uses cssjanus under the hood to flip style orientations for RTL",
-  "main": "dist/stylis-rtl.js",
+  "main": "dist/lib/stylis-rtl.js",
```
Comment (probablyup) [inline package.json:5]: let's keep the directory flat and just do `stylis-rtl.cjs.js` and `stylis-rtl.esm.js`

Comment (layershifter) [reply package.json]: image

Done 👍

File: .babelrc (hunk 1 above)
Comment (layershifter) [reply .babelrc]: Bumped. Should I also add `engines` entry to `package.json`?

Comment (probablyup) [reply .babelrc]: Yes please

File: package.json [hunk 3]
```diff
@@ -41,16 +47,16 @@
     "stylis": "4.x"
   },
   "devDependencies": {
-    "@babel/cli": "^7.0.0",
-    "@babel/core": "^7.0.0",
-    "@babel/preset-env": "^7.0.0",
-    "@babel/preset-flow": "^7.0.0",
-    "babel-core": "^7.0.0-bridge.0",
```
Comment (layershifter) [inline package.json:48]: These is no sense to have `babel-core` as Babel v7 is used

File: .babelrc (hunk 1 above)
Comment (layershifter) [reply .babelrc]: Added, please check

**Comment by probablyup** [comment]
@layershifter have you tested this with a project?

**Pull request closed without merging**